### PR TITLE
refactor: Removed container orchestration bundles from Docker images.

### PR DESCRIPTION
**Pull request by salvatore-coppola**
> **Note**: We are using the Conventional Commits convention for our pull request titles. Please take a look at the [PR title format document](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#submitting-the-changes) for the supported [types](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#type) and [scopes](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#scope).

Removed container orchestration bundles from Docker images and created a new docker specific profile.

### CODE REVIEW

File: kura/distrib/src/main/ant/build_equinox_distrib.xml [hunk 1]
```diff
@@ -111,6 +111,9 @@
 		<condition property="is.nn">
 			<contains string="${build.name}" substring="-nn" />
 		</condition>
+        <condition property="is.docker">
```
Comment (marcellorinaldo) [inline kura/distrib/src/main/ant/build_equinox_distrib.xml:114]: Please fix indentation.

File: kura/distrib/src/main/resources/docker-x86_64-nn/jdk.dio.properties [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#
+#  Copyright (c) 2011, 2021 Eurotech and/or its affiliates and others
```
Comment (marcellorinaldo) [inline kura/distrib/src/main/resources/docker-x86_64-nn/jdk.dio.properties]: Where does this mapping come from? I don't think it makes sense to have such mappings on a generic container.

File: kura/distrib/src/main/resources/docker-x86_64-nn/kura_install.sh [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+find /opt/eclipse/kura -type d -exec chmod u+x "{}" \;
+
+#enable kurad user to HALT functionality
+for group in gpio leds spi i2c; do
+    if ! getent group $group >/dev/null; then
+        addgroup --system $group
+    fi
+done
+
+usermod -a -G gpio kurad
```
Comment (marcellorinaldo) [inline kura/distrib/src/main/resources/docker-x86_64-nn/kura_install.sh]: Are all those commands necessary on a container?

File: kura/distrib/src/main/resources/docker-x86_64-nn/kura_install.sh [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+# set up kura files permissions
+chmod 700 ${INSTALL_DIR}/kura/bin/*.sh
+chown -R kurad:kurad /opt/eclipse
+chmod -R go-rwx /opt/eclipse
+chmod a+rx /opt/eclipse    
+find /opt/eclipse/kura -type d -exec chmod u+x "{}" \;
+
+#enable kurad user to HALT functionality
+for group in gpio leds spi i2c; do
```
Comment (marcellorinaldo) [inline kura/distrib/src/main/resources/docker-x86_64-nn/kura_install.sh]: Does it make sense to modify gpio, leds, and i2c groups on a container?

**Review by marcellorinaldo — CHANGES_REQUESTED** [review]
The alpine build fails:

```
#8 687.0      [move] Warning: Could not find file /kura/kura/distrib/target/plugins/org.eclipse.kura.protocol.can.jar to copy.
#8 687.7      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/raspberry-pi-ubuntu-20/selinuxKura.pp to copy.
#8 786.6      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/raspberry-pi-ubuntu-20-nn/selinuxKura.pp to copy.
#8 872.4      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/raspberry-pi/selinuxKura.pp to copy.
#8 974.7      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/raspberry-pi-nn/selinuxKura.pp to copy.
#8 1062.7      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/intel-up2-ubuntu-20/selinuxKura.pp to copy.
#8 1163.4      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/intel-up2-ubuntu-20-nn/selinuxKura.pp to copy.
#8 1247.5      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/nvidia-jetson-nano/selinuxKura.pp to copy.
#8 1347.9      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/nvidia-jetson-nano-nn/selinuxKura.pp to copy.
#8 1503.9 /bin/sh: /kura/kura/distrib/target/kura_*_docker-x86_64-nn_installer.sh: not found
```

File: kura/distrib/src/main/ant/build_equinox_distrib.xml (hunk 1 above)
Comment (salvatore-coppola) [reply kura/distrib/src/main/ant/build_equinox_distrib.xml:114]: i see the indentation in the right way, however i'll make a commit replacing all tabs with whitespaces to be aligned with other files.

**Comment by salvatore-coppola** [comment]
> The alpine build fails:
> 
> ```
> #8 687.0      [move] Warning: Could not find file /kura/kura/distrib/target/plugins/org.eclipse.kura.protocol.can.jar to copy.
> #8 687.7      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/raspberry-pi-ubuntu-20/selinuxKura.pp to copy.
> #8 786.6      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/raspberry-pi-ubuntu-20-nn/selinuxKura.pp to copy.
> #8 872.4      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/raspberry-pi/selinuxKura.pp to copy.
> #8 974.7      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/raspberry-pi-nn/selinuxKura.pp to copy.
> #8 1062.7      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/intel-up2-ubuntu-20/selinuxKura.pp to copy.
> #8 1163.4      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/intel-up2-ubuntu-20-nn/selinuxKura.pp to copy.
> #8 1247.5      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/nvidia-jetson-nano/selinuxKura.pp to copy.
> #8 1347.9      [copy] Warning: Could not find file /kura/kura/distrib/src/main/resources/nvidia-jetson-nano-nn/selinuxKura.pp to copy.
> #8 1503.9 /bin/sh: /kura/kura/distrib/target/kura_*_docker-x86_64-nn_installer.sh: not found
> ```

the last build works: https://github.com/eclipse/kura/pull/4178/commits/1d9725124d63378f0c09aa347df9b785de5983a4

**Comment by marcellorinaldo** [comment]
> the last build works: https://github.com/eclipse/kura/commit/1d9725124d63378f0c09aa347df9b785de5983a4

Yes, the CI does not build the containers from distrib, just the deb packages. If you try to build the container it will fail:

```shell
cd kura/container/<kura_alpine or kura_centos>
docker build -t <tag> .
```

**Comment by salvatore-coppola** [comment]
> > the last build works: [1d97251](https://github.com/eclipse/kura/commit/1d9725124d63378f0c09aa347df9b785de5983a4)
> 
> Yes, the CI does not build the containers from distrib, just the deb packages. If you try to build the container it will fail:
> 
> ```shell
> cd kura/container/<kura_alpine or kura_centos>
> docker build -t <tag> .
> ```

you are right, i'll fix that.

**Comment by salvatore-coppola** [comment]
I've tested the container buidling on my machine due the build mechanism using my branch on my fork and everything seems to work fine with the following command:

`docker build --build-arg GIT_REPO=https://github.com/salvatore-coppola/kura.git --build-arg GIT_BRANCH=remove-container-orchestator-from-docker-image -t kura_alpine .`

**Review by marcellorinaldo — COMMENTED** [review]
Why has `target-platform/org.eclipse.kura.libsocket-can-osgi/lib/libsocket-can-java.jar` been added?

**Comment by salvatore-coppola** [comment]
> `target-platform/org.eclipse.kura.libsocket-can-osgi/lib/libsocket-can-java.jar`

building leftover.

**Review by marcellorinaldo — APPROVED** [review]

**Pull request merged**